### PR TITLE
Add initial main functionality and tests

### PR DESCRIPTION
**Pull request by kmuncie**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -3,7 +3,41 @@
...
    return {
-      postcssPlugin: 'postcss-var-fallback',
+      postcssPlugin: 'postcss-css-var-fallback',
+
+      Root(root) {
+         root.walkDecls((decl) => {
+            // Capture declarations with include var()
+            const regex = new RegExp(/var\(/);
+
+            var getVarFallback = function(value) {
```
Comment (kmuncie) [inline index.js:13]: Linting is failing on this but I am not sure the right way to refactor this. I am sure more of this file can be improved and I am eager to learn.

Comment (pbredenberg) [reply index.js:13]: Have a look at the linter output.

https://app.travis-ci.com/github/silvermine/postcss-css-var-fallback/jobs/537962832#L222

Our standards dictate that a variable definition not span multiple lines. A better approach might be to move the function to the returned object level:

```
return {
   myFunction: (value) => {
     // stuff
   },
   
   Root(root) {
     root.walk(() => {
      // stuff
      this.myFunction(arg);
    }
   }
}
```

Does that make sense?

**Review by pbredenberg — CHANGES_REQUESTED** [review]
Just a couple of thing here, let me know if you'd like to pair.

File: index.js (hunk 1 above)
Comment (pbredenberg) [reply index.js]: As discussed IRL we had to move the function into the `Root` hook due to object scope.

Comment (kmuncie) [reply index.js]: ADIRL I couldn't move it outside but we improved things with your help

File: index.js [hunk 2]
```diff
@@ -3,7 +3,47 @@
 module.exports = () => {
 
    return {
-      postcssPlugin: 'postcss-var-fallback',
+      postcssPlugin: 'postcss-css-var-fallback',
+
+
+      Root(root) {
+         root.walkDecls((decl) => {
+
```
Comment (pbredenberg) [inline index.js:11]: Sorry for the nit, but I'd lose one of these extra lines

**Review by pbredenberg — CHANGES_REQUESTED** [review]
SORRY! Just one thing. All the rest look good

File: index.js (hunk 2 above)
Comment (kmuncie) [reply index.js:10]: No ofcourse, fixing now

**Comment by kmuncie** [comment]
@pbredenberg This look okay now?

**Review by pbredenberg — APPROVED** [review]
Ooooohh! Pretty!

**Pull request merged**